File: app/src/main/java/blue/koenig/kingsfinances/features/statistics/AssetsCalculatorService.java
package blue.koenig.kingsfinances.features.statistics;

import com.koenig.commonModel.finance.BankAccount;

import org.joda.time.DateTime;

import java.util.List;
import java.util.Map;

import blue.koenig.kingsfinances.model.calculation.StatisticEntry;

/**
 * Created by Thomas on 05.01.2018.
 */

public interface AssetsCalculatorService {
    Map<BankAccount, List<StatisticEntry>> loadAllBankAccountStatistics();

    DateTime getStartDate();

    DateTime getEndDate();

    void save(Map<BankAccount, List<StatisticEntry>> statisticEntryLists);
}

File: app/src/main/java/blue/koenig/kingsfinances/features/statistics/AssetsCalculatorService.kt
package blue.koenig.kingsfinances.features.statistics

import blue.koenig.kingsfinances.model.calculation.StatisticEntryDeprecated
import com.koenig.commonModel.finance.BankAccount
import org.joda.time.DateTime

/**
 * Created by Thomas on 05.01.2018.
 */

interface AssetsCalculatorService {

    val startDate: DateTime

    val endDate: DateTime

    val overallString: String

    val futureString: String
    fun loadAllBankAccountStatistics(): Map<BankAccount, List<StatisticEntryDeprecated>>

    fun save(statisticEntryLists: Map<BankAccount, List<StatisticEntryDeprecated>>)
}
